File: dev/20190531/session/api/fizzbuzz/src/main/java/fizzbuzz/FizzBuzzType03.java
package fizzbuzz;

public class FizzBuzzType03
  extends IFizzBuzzType {

  public FizzBuzzValue generate(int number) {
    boolean isFizz = number % 3 == 0;
    boolean isBuzz = number % 5 == 0;
    if (isFizz && isBuzz) {
      return new FizzBuzzValue(number, IFizzBuzzType.FIZZ_BUZZ);
    }
    return new FizzBuzzValue(number, Integer.toString(number));
  }

}

File: dev/20190822/k2works/src/FizzBuzzType03.ts
import { FizzBuzz } from "./FizzBuzz";
import { FizzBuzzValue } from "./FizzBuzzValue";
import { IFizzBuzzType } from "./IFizzBuzzType";

export class FizzBuzzType03 extends IFizzBuzzType {
  public generate(input: number): FizzBuzzValue {
    const isFizz: boolean = input % 3 === 0;
    const isBuzz: boolean = input % 5 === 0;

    if (isFizz && isBuzz) {
      return new FizzBuzzValue(input, IFizzBuzzType.FIZZ_BUZZ);
    }
    return new FizzBuzzValue(input, input.toString());
  }
}
